opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -273,6 +286,103 @@ class Tracer(ABC):
     and controlling spans' lifecycles.
     """
 
+    def decorate(
+        self,
+        name: str = None,
+        context: Optional[Context] = None,
+        kind: SpanKind = SpanKind.INTERNAL,
+        attributes: types.Attributes = None,
+        links: _Links = None,
+        start_time: Optional[int] = None,
+        record_exception: bool = True,
+        set_status_on_exception: bool = True,
+        end_on_exit: bool = True,
+        span_arg_name: str | None = "span",
```

Comment:
Would it make sense for this to be implemented by its own decorator? It could be a free function which is nice and could be stacked with other decorators if needed
```py
@trace.start_as_current_span("foo")
@with_span_arg("the_span")
def foo(the_span: Span):
  ...
```

Reply:
You can open an issue if you think it's a useful feature.

I have tested the `get_current_span().set_attribute(...)` and I think it's a better way to make it works without surcharging the function parameters that must be more business oriented...